input.go

```diff
@@ -264,3 +266,32 @@ type eventFirer interface {
 }
 
 // -------------------------------------------------------------------------------------
+
+type inputManager struct {
+	tempItems []Shape
+	g         *Game
+	id2Timer  map[engine.Object]int64
+}
+
+const (
+	// minimum interval between two mouse click events
+	inputMouseClickIntervalMs = 50
+)
+
+func (p *inputManager) init(g *Game) {
+	p.tempItems = make([]Shape, 50)
+	p.id2Timer = make(map[engine.Object]int64)
+	p.g = g
+}
+
+func (p *inputManager) canTriggerClickEvent(id engine.Object) bool {
```

Comment:
If the user clicks and holds for a long time, will the `click` event trigger multiple times?

In SPX v1, it will only trigger once when pressed.

Reply:
`If the user clicks and holds for a long time, will the click event trigger multiple times?`
No, it would only trigger once